modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -530,7 +530,12 @@ private void generateApis(List<File> files, List<Object> allOperations, List<Obj
         }
         Map<String, List<CodegenOperation>> paths = processPaths(this.openAPI.getPaths());
         Set<String> apisToGenerate = null;
-        String apiNames = GlobalSettings.getProperty("apis");
+        String apiNames = GlobalSettings.getProperty(CodegenConstants.APIS);
+
+        if (apiNames == null || StringUtils.isBlank(apiNames)) {
+            apiNames = (String) opts.getConfig().additionalProperties().get(CodegenConstants.APIS);
```

Comment:
I don't think this will work… it would mean that generators can no longer support an `apis` additional property for other reasons as any setting here would cause no API classes to get generated.

Can you provide an example of how you've configured the Maven plugin? I think it would be best to fix system property support there.